### DISCUSSION TITLE
Google Authentication isssue providerAccountId is undefined after successful login

### DISCUSSION THREAD

**Discussion by laurent512** [nextauthjs/next-auth#10891] [Help]
Hello,

I've encountered an issue with OAuth login functionality on my website. I've implemented both GitHub and Google login methods. While GitHub login is working smoothly, I'm facing a peculiar problem with Google login.

Here's the scenario:

After clearing all user, session, and account tables in the database, I log in using my Gmail account 'A', and everything works as expected. A new user, account, and session are created successfully.

Next, I sign out and attempt to log in with Gmail account 'B'. Upon selection of Gmail account 'B' I am redirected back to my website somehow successfully, but here's the issue: it identifies me as Gmail account 'A'. 
From database point of view, it created a new session for the same user 'A', which is quite unexpected. (also no new account or user is created)

What I have observed also is that the "providerAccountId" in the account table is undefined (maybe leading to wrong equivalence..,)

Has anyone else encountered this issue with the latest beta version of NextAuth.js? I'm using Drizzle, NextAuth, and Next.js for my application.

Your insights would be greatly appreciated :)

**Comment by laurent512** [comment]
My theory is correct, problem is that providerAccountId is not populated with my google provider. 
This leading to undefined == undefined. 

I checked that by recreating the process, but after 'A' sign in , I wrote manually in the database  a random string in the providerAccountId field. 
When logging with 'B' , it random string was not equal to undefined so new account has been created. (but this new one has undefined in providerAccountId obviously) 

Any guess where this might come?

**Comment by laurent512** [comment]
From debug mode we can see this : 
![image](https://github.com/nextauthjs/next-auth/assets/4173291/69b04cf6-4f6d-4f87-b300-138af5117667)

**Comment by laurent512** [comment]
Weirdly enough , this is related to calling profile()
![image](https://github.com/nextauthjs/next-auth/assets/4173291/0eb8eca2-cb16-461d-a97f-2a6efbb93f86)
When I don't call it, everything is working well, as soon as I include it, the bug starts appearing.

**Reply by radioac7iv3** [reply]
Hmm I see, were you able to figure it out, why this could be happening, in my case the providerAccountId is changing on every consecutive login, not sure why
